### PR TITLE
Improve RTN15G tests

### LINKED ISSUE #725: Improve RTN15G tests
- [x] Improve usage of reconnection delay See https://github.com/ably/ably-ios/pull/719#discussion_r188201480
- [x] Disconnect after attaching See https://github.com/ably/ably-ios/pull/719#discussion_r188201665
- [x] Add check for `resume` See https://github.com/ably/ably-ios/pull/719#discussion_r188202070
- [x] Use standard connection values when possible See https://github.com/ably/ably-ios/pull/719#discussion_r188202285

### PR DESCRIPTION
**Pull request by funkyboy**
Fix #725

### CODE REVIEW

**Comment by mattheworiordan** [comment]
@funkyboy what about https://github.com/ably/ably-ios/pull/719#discussion_r188307367?  Do you have any thoughts on why the client is not automatically reconnecting, or am I reading the test incorrectly?

**Comment by funkyboy** [comment]
@mattheworiordan [Looking at this](https://github.com/ably/ably-ios/blob/67d1d943ceba3181a035df8e599fb9b10a5fd618/Source/ARTRealtime.m#L489) I am quite sure it only uses the timer and does not honor the "will immediately attempt to reconnect and restore the connection state" part of [RTN15a](https://docs.ably.io/client-lib-development-guide/features/#RTN15a)

Will check with @ricardopereira.

**Comment by mattheworiordan** [comment]
Yes, looking at https://github.com/ably/ably-ios/blob/ec182697eb4842f67eb28a30bf80eb038f8545ad/Spec/RealtimeClientConnection.swift#L2212-L2263, which is supposedly testing `RTN15a`, yet it's not, there is no test for this.

**Comment by funkyboy** [comment]
filed an issue about RTN15a here https://github.com/ably/ably-ios/issues/727 
Can we merge this?

@mattheworiordan @paddybyers

**Comment by mattheworiordan** [comment]
@funkyboy yes I think so 👍

File: Spec/RealtimeClientConnection.swift [hunk 1]
```diff
@@ -2706,18 +2706,19 @@ class RealtimeClientConnection: QuickSpec {
...
+                                            channel.once(.attached) { stateChange in
+                                                expect(stateChange?.resumed).to(beFalse())
                                                 done()
                                             }
                                         }
                                     }
                                 }
-                                client.onDisconnected()
                             }
                         }
```
Comment (ricardopereira) [inline Spec/RealtimeClientConnection.swift]: I think the `client.connect()` should be called in the `waitUntil` closure. Just to be sure that all observers are set.

Comment (funkyboy) [reply Spec/RealtimeClientConnection.swift]: @ricardopereira good point :) See https://github.com/ably/ably-ios/pull/726/commits/f0e2314b37160152a6a1af690fbc3167b0970831

**Pull request merged**